src/components/reservation/reservation-modal.tsx

```diff
@@ -107,67 +123,73 @@ const ReservationModal = ({ manifestation }: ReservationModalProps) => {
       )}
       closeModalAriaLabelText={t("reservationModalCloseModalAriaLabelText")}
     >
-      {reservationDidSuccess && patron && (
+      {reservationResult === "success" && reservationDetails && patron && (
         <ReservationSucces
           modalId={reservationModalId(faustId)}
           title={titles.main[0]}
           preferredPickupBranch={getPreferredLocation(
             patron.preferredPickupBranch,
             branchData
           )}
+          numberInQueue={String(reservationDetails.numberInQueue)}
         />
       )}
 
-      {reservationDidSuccess === false && (
-        <ReservationError setReservationDidSuccess={setReservationDidSuccess} />
+      {reservationResult === "already_reserved" && (
+        <ReservationError
+          setReservationResponse={setReservationResponse}
+          errorDescription={t("alreadyReservedText")}
+          buttonText={t("closeText")}
+        />
       )}
 
-      {reservationDidSuccess === null && (
-        <section className="reservation-modal">
-          <header className="reservation-modal-header">
-            <Cover pid={pid as Pid} size="medium" animate />
-            <div className="reservation-modal-description">
-              <div className="reservation-modal-tag">
-                {materialTypes[0].specific}
+      {reservationResult !== "success" &&
+        reservationResult !== "already_reserved" && (
```

Comment:
Please revisit this. As mentioned there are many more results apart from "success" and "already_reserved" which would cause the modal to appear again. As I see it we only want to display the form if there no request has been made.

Also consider moving the form into a separate component like we have for error and success. The modal component is growing quite large.